src/Microsoft.Dnx.Tooling/NuGet/Core/Configuration/Settings.cs

```diff
@@ -774,7 +775,23 @@ private void ExecuteSynchronized(Action ioOperation)
             var fileName = _fileSystem.GetFullPath(_fileName);
 
             // Global: ensure mutex is honored across TS sessions 
-            using (var mutex = new Mutex(false, "Global\\" + EncryptionUtility.GenerateUniqueToken(fileName)))
+            var mutexName = $"Global\\{EncryptionUtility.GenerateUniqueToken(fileName)}";
+            Mutex mutex = null;
+#if DNXCORE50
+            if (RuntimeEnvironmentHelper.IsWindows)
+            {
+                mutex = new Mutex(false, mutexName);
+            }
+            else
+            {
+                // However, named mutex is not supported on cross-plat CoreCLR
+                // so we fall back to process-local synchronization
+                mutex = new Mutex(false);
```

Comment:
Don't you need a lookup mechanism by name here too, like for the Semaphore case? Otherwise it looks like it would create a new Mutex every time and no two calls would share a mutex to synchronize file access.